language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -344,8 +344,10 @@ data TemplateDef = TemplateDef
   , tplPkgId :: PackageId
   , tplModule :: ModuleName
   , tplDecoder :: Decoder
+  , tplEncode :: Encode
   , tplKeyDecoder :: Maybe Decoder
   -- ^ Nothing if we do not have a key.
+  , tplKeyEncode :: Encode
```

Comment:
`tplKeyDecoder` is a `Maybe`. Why is `tplKeyEncode` not a `Maybe` as well instead of the slightly weird `EncodeThrow` thing?

Reply:
`tplKeyDecoder` is a `Maybe`, but the first thing we do with it is to turn it into a `(T.Text, Decoder)` where the decoder is `DecoderConstant ConstantUndefined`, and while `DecoderConstant` is used in some other cases, `ConstantUndefined` only ever gets created here, so this feels like unnecessary juggling to me. It would have seemed more straightforward to have a `Decoder` case for that, so that's what I did for `Encode`. Happy to switch, though.

The `Maybe` would have made sense to me if the goal was to not produce the `keyDecoder` entry if it's `None`.